SONIC-DEV/sonic-buildimage
# [platform/Ali] Add optical module I2C bus recovery.

**- What I did**
Add the feature to detect the Optical Module I2C bus recovery on Fishbone32, Fishbone48, and Phalanx. This prevent whole I2C bus lockup when a single optical module is hanging up.

**- How I did it**
Update the BSP driver to detect the bus stuck when doing I2C transfer.
If the bus stuck found, report error to kernel log. Then trying to recovery by resetting the I2C switch chip.

**- How to verify it**
1. Simulate the optical module bus stuck by pull `SDA` of SFP port1 to logic low.
2. Try to access the port using `sfputil show eeprom -p Ethernet0`
3. Check the kernel log for error report.
4. Try to access other port I2C `sfputil show eeprom -p Ethernetx` with success.
5. The bus stuck must not block other port I2C access.

**- Description for the changelog**


*Add optical module I2C bus recovery on Fishbone32, Fishbone48, and Phalanx.

**- A picture of a cute animal (not mandatory but encouraged)**
